src/telemetry/telemetry.ts

```diff
@@ -30,20 +30,20 @@ class Telemetry implements ITelemetry {
   }
 
   public trackEvent(eventName: string, payload: any) {
-    this.appInsights.trackEvent({ name: eventName }, payload);
+    this.appInsights.trackEvent({ name: eventName, properties: payload });
   }
 
   public trackException(error: Error, severityLevel: SeverityLevel) {
     this.appInsights.trackException({ error, severityLevel });
   }
 
-  public trackReactComponent(ComponentToTrack: ComponentType): ComponentType {
-    return withAITracking(this.reactPlugin, ComponentToTrack);
+  public trackReactComponent(ComponentToTrack: ComponentType, componentName?: string): ComponentType {
+    return withAITracking(this.reactPlugin, ComponentToTrack, componentName);
   }
 
   private filterFunction(envelope: any) {
     // Identifies the source of telemetry events
-    envelope.baseData.name = 'Graph Explorer v4';
+    envelope.tags['ai.cloud.role'] = 'Graph Explorer v4';
```

Comment:
Why are we using the cloud role for this name?  Normally that means more like it's a front end machine or something.

Reply:
I borrowed the idea from the fact that the property could be used in a microservices environment to identify individual services.  

To remove confusion, I've added application name as a custom property to be able to properly identify GE insights.